File: Human_written_code/Code_with_comments/Python/Q24.py
def main():
    # Read input string and remove any leading or trailing whitespace
    s1 = input().strip()
    
    # Initialize a counter to track consecutive identical characters
    condition = 1

    # Iterate through the string from the second character onward
    for i in range(1, len(s1)):
        # Check if the current character is the same as the previous one
        if s1[i] == s1[i - 1]:
            # Increase the consecutive character counter
            condition += 1
            # If we reach 7 consecutive identical characters, print "YES" and exit
            if condition == 7:
                print("YES")
                return
        else:
            # Reset the counter if the characters are not consecutive
            condition = 1

    # If no sequence of 7 consecutive identical characters is found, print "NO"
    print("NO")

# Entry point of the script
if __name__ == "__main__":
    main()

File: Human_written_code/Code_with_comments/CPP/Q24.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    string s;
    int d = 1;  // Counter for consecutive identical characters

    cin >> s;

    for (int i = 0; i < s.length() - 1; ++i) {
        if (d >= 7)  // Stop if 7 consecutive identical characters are found
            break;

        if (s[i] == s[i + 1]) {
            d++;  // Increment count if characters match
        } else {
            d = 1;  // Reset count if characters differ
        }
    }

    // Output result based on the count of consecutive identical characters
    cout << (d >= 7 ? "YES" : "NO");

    return 0;
}
